### PR TITLE
refactor: balance fetching via paraspell

### PR DESCRIPTION
**Pull request by noahjoeris**

### CODE REVIEW

File: app/src/hooks/useBalance.tsx [hunk 1]
```diff
@@ -62,24 +61,18 @@ const useBalance = ({ env, api, chain, token, address }: UseBalanceParams) => {
...
-              formatted: toHuman(result?.free || 0n, token).toString(),
-              decimals: token.decimals,
-              symbol: token.symbol,
-            }
+          const relay = getRelayNode(env)
+          const node = assets.getTNode(chain.chainId, relay)
+          if (!node) throw new Error('Node not found')
+
+          const tokenSymbol = getTokenSymbol(node, token)
+          const balance = await getAssetBalance(address, node, { symbol: tokenSymbol })
```
Comment (NunoAlexandre) [inline app/src/hooks/useBalance.tsx:69]: NIT but I'd add the fallback here so you don't have to repeat it down below
```suggestion
          const balance = (await getAssetBalance(address, node, { symbol: tokenSymbol })) ?? 0
```

**Review by NunoAlexandre — APPROVED** [review]
Looks good :) just giving it a try

**Review by NunoAlexandre — APPROVED** [review]
✨

**Review by Victor-Salomon — APPROVED** [review]
It looks good to me 😊

**Pull request merged**